amethyst_core/src/transform/components/transform.rs

```diff
@@ -560,6 +561,31 @@ impl Transform {
             .to_homogeneous()
             .append_nonuniform_scaling(&inv_scale)
     }
+
+    /// Calculates the inverse of this transform, which is in effect the 'view matrix' as
+    /// commonly seen in computer graphics. This function computes the view matrix for the
+    /// global transformation of the entity, and so takes into account Parents.
+    ///
+    /// We can exploit the extra information we have to perform this inverse faster than `O(n^3)`.
+    pub fn global_view_matrix(&self) -> Matrix4<Float> {
+        // TODO: check if this actually is faster
+        let mut res = self.global_matrix.clone();
+
+        {
+            let mut slice3x3 = res.fixed_slice_mut::<na::U3, na::U3>(0, 0);
+            assert!(slice3x3.try_inverse_mut());
+        }
+
+        let mut translation = -res.column(3).xyz();
+        translation = res.clone().fixed_slice::<na::U3, na::U3>(0, 0) * translation;
+
+        let mut res_trans = res.column_mut(3);
```

Comment:
This can probably be written as
```rust
res.column_mut(3).copy_from(translation);
```

Reply:
This does not work and causes a compile error.

`error[E0277]: the trait bound `nalgebra::base::constraint::ShapeConstraint: nalgebra::base::constraint::SameNumberOfRows<nalgebra::base::dimension::U4, nalgebra::base::dimension::U3>` is not satisfied
   --> amethyst_core/src/transform/components/transform.rs:579:41
    |
579 |         translation = res.column_mut(3).copy_from(&translation);
    |                                         ^^^^^^^^^ the trait `nalgebra::base::constraint::SameNumberOfRows<nalgebra::base::dimension::U4, nalgebra::base::dimension::U3>` is not implemented for `nalgebra::base::constraint::ShapeConstraint`
    |
    = help: the following implementations were found:
              <nalgebra::base::constraint::ShapeConstraint as nalgebra::base::constraint::SameNumberOfRows<D, D>>
              <nalgebra::base::constraint::ShapeConstraint as nalgebra::base::constraint::SameNumberOfRows<D, nalgebra::base::dimension::Dynamic>>
              <nalgebra::base::constraint::ShapeConstraint as nalgebra::base::constraint::SameNumberOfRows<nalgebra::base::dimension::Dynamic, D>>

error[E0308]: mismatched types
   --> amethyst_core/src/transform/components/transform.rs:579:23
    |
579 |         translation = res.column_mut(3).copy_from(&translation);
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `nalgebra::base::matrix::Matrix`, found ()
    |
    = note: expected type `nalgebra::base::matrix::Matrix<float::Float, nalgebra::base::dimension::U3, nalgebra::base::dimension::U1, nalgebra::base::array_storage::ArrayStorage<float::Float, nalgebra::base::dimension::U3, nalgebra::base::dimension::U1>>`
               found type `()`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
`

The current version is already about 10x faster.